extremeflash/ws_ap3935i.py

```diff
@@ -1,232 +0,0 @@
-import ipaddress
-import logging
-import pathlib
-import re
-import time
-from threading import Thread
-from typing import Optional
-
-import serial
-
-from .tftp_server import TftpServer
-from . import ws_ap3710i
-from .ws_ap3710i import (
-    boot_set_ips,
-    boot_wait_for_brlan,
-    bootup_interrupt,
-    bootup_login,
-    bootup_login_verification,
-    debug_serial,
-    is_kernel_booting,
-    keep_logging_until_reboot,
-    post_cleanup,
-    readline_from_serial,
-    setting_up_ips,
-    start_ssh,
-    write_to_serial,
-)
-
-#  See .ws_ap3710i file for implementation details
-
-
-def bootup_set_boot_openwrt(ser: serial.Serial):
-    if not ws_ap3710i.event_keep_serial_active.is_set():
-        return
-    ser.write(b"printenv\n")
-    time.sleep(1)
-    printenv_return = ser.read(ser.in_waiting).decode("ascii")
-    debug_serial(printenv_return)
-
-    if "MODEL=AP3935i" not in printenv_return:
-        model = re.search(r"MODEL=(.*)\r\n", printenv_return)
-        raise RuntimeWarning(
-            f"Unexpected Model {None if model is None else model.group(1)} found. Aborting to not harm device."
-        )
-
-    # https://git.openwrt.org/?p=openwrt/openwrt.git;a=commit;h=3aef61060e3f51aa43fe494d5ff173e81dd43003
-    # following parameters works. Not found in the above commit.
-    boot_openwrt_params = b"sf probe 0; sf read 0x41500000 0x003c0000 0x00e10000; bootm 0x41500000"
-
-    if "boot_openwrt" in printenv_return:
-        logging.debug("Found existing U-Boot boot_openwrt parameter. Verifying.")
-        existing_boot_openwrt_params = re.search(r"boot_openwrt=(.*)\r\n", printenv_return)
-        if not existing_boot_openwrt_params:
-            raise RuntimeError("Unable to parse detected boot_openwrt paramter")
-
-        if boot_openwrt_params.decode("ascii") != existing_boot_openwrt_params.group(1):
-            # Some AP3825i had wrong and/or outdated boot_openwrt parameters in the past.
-            logging.warning(f"Overwriting unexpected param for 'boot_openwrt': {existing_boot_openwrt_params.group(0)}")
-        else:
-            # do not set anything if we found boot_openwrt
-            # TODO: should we check if bootcmd is also set correctly?
-            logging.debug("Existing U-Boot boot_openwrt parameter looks good.")
-            return
-    else:
-        logging.info("Did not find boot_openwrt in U-Boot parameters. Setting it.")
-
-    write_to_serial(ser, b'setenv boot_openwrt "' + boot_openwrt_params + b'"\n')
-    time.sleep(0.5)
-
-    write_to_serial(ser, b'setenv bootcmd "run boot_openwrt"\n')
-    time.sleep(0.5)
-
-    if ws_ap3710i.DRYRUN:
-        logging.info("dryrun: Skipping saveenv")
-        return
-
-    ser.write(b"saveenv\n")
-    time.sleep(2)
-    saveenv_return = ser.read(ser.in_waiting).decode("ascii")
-    debug_serial(saveenv_return)
-
-    if "Writing to NAND" not in saveenv_return and "Writing to redundant NAND" not in saveenv_return:
```

Comment:
This section seems to be missing completely in the refactored code.

Reply:
True, is it okay to check for either of the three strings:
- "Writing to Flash"
- "Writing to NAND"
- "Writing to redundant NAND"

or does the first one exist with the 3935i, but the others are required?
Or would you rather have this separate by device?

Reply:
Should be fine. It would surprise me if a device has both Flash and NAND at the same time.